pom.xml

```diff
@@ -310,6 +332,57 @@
+                <artifactId>groovy-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>update-github-workflows</id>
+                        <inherited>false</inherited>
+                        <goals>
+                            <goal>execute</goal>
+                        </goals>
+                        <phase>validate</phase>
+                        <configuration>
+                            <source>
+                                import java.nio.file.Files
+                                import java.nio.file.Path
+                                import java.nio.file.Paths
+
+                                def buildYaml = '.github/workflows/build.yml'
+                                def buildXmlPath = project.basedir.toPath().resolve(buildYaml)
+                                def buildXmlSource = buildXmlPath.getText('UTF-8')
+                                def pattern = java.util.regex.Pattern.compile('testModule: *\\[([^\\]]*)\\]')
+                                def matcher = pattern.matcher(buildXmlSource)
+                                if (!matcher.find()) {
+                                    throw new IllegalStateException('Pattern ' + pattern + ' does not match ' + buildXmlPath)
+                                }
+                                def foundTestModules = matcher.group(1)
+
+                                def expectedTestModules;
+                                def iTestsDir = project.basedir.toPath().resolve('integration-tests')
+                                try (java.util.stream.Stream files = Files.list(iTestsDir)) {
+                                    expectedTestModules = files
+                                        .filter(path -> Files.isRegularFile(iTestsDir.resolve(path).resolve('pom.xml')))
+                                        .map(Path::getFileName)
+                                        .map(Path::toString)
+                                        .sorted()
+                                        .collect(java.util.stream.Collectors.joining(', '))
+                                }
+                                if (!foundTestModules.equals(expectedTestModules)) {
+                                    println('Updating testModule attribute in ' + buildYaml + ':\n  old: ' + foundTestModules + '\n  new: '+expectedTestModules)
+                                    def newSource = new StringBuilder()
+                                    matcher.appendReplacement(newSource, 'testModule: ['+ expectedTestModules +']')
+                                    matcher.appendTail(newSource)
+                                    buildXmlPath.setText(newSource.toString(), 'UTF-8')
+                                } else {
+                                    println('testModule attribute in ' + buildYaml + ' is up to date')
+                                }
+                            </source>
```

Comment:
@ppalaga to me this is not self-explaining at first glance, so could you please add a small description what this does (next time you touch this)? Thanks!